lib/compat/wordpress-6.1/class-wp-theme-json-resolver-6-1.php

```diff
@@ -188,4 +201,35 @@ public static function get_user_data_from_wp_global_styles( $theme, $create_post
 
 		return $user_cpt;
 	}
+
+	/**
+	 * Checks whether the registered blocks were already processed for this origin.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $origin Data source for which to cache the blocks.
+	 *                       Valid values are 'core', 'blocks', 'theme', and 'user'.
+	 * @return bool True on success, false otherwise.
+	 */
+	protected static function has_same_registered_blocks( $origin ) {
+		// Bail out if the origin is invalid.
+		if ( ! isset( static::$blocks_cache[ $origin ] ) ) {
+			return false;
+		}
+
+		$registry = WP_Block_Type_Registry::get_instance();
+		$blocks   = $registry->get_all_registered();
+
+		// Is there metadata for all currently registered blocks?
+		$block_diff = array_diff_key( $blocks, static::$blocks_cache[ $origin ] );
+		if ( empty( $block_diff ) ) {
+			return true;
+		}
+
+		foreach ( $blocks as $block_name => $block_type ) {
+			static::$blocks_cache[ $origin ][ $block_name ] = true;
+		}
+
+		return false;
+	}
 }
```

Comment:
Also need to add the changes for `WP_Theme_JSON_Resolver::get_theme_data()`. Why? Gutenberg's class extends from `WP_Theme_JSON_Resolver_6_0` which will overload Core's `get_theme_data()` method, meaning Core's won't be used. That can cause unexpected differences in the way the plugin works vs Core.

Reply:
```suggestion

	/**
	 * Returns the theme's data.
	 *
	 * Data from theme.json will be backfilled from existing
	 * theme supports, if any. Note that if the same data
	 * is present in theme.json and in theme supports,
	 * the theme.json takes precedence.
	 *
	 * @since 5.8.0
	 * @since 5.9.0 Theme supports have been inlined and the `$theme_support_data` argument removed.
	 * @since 6.0.0 Added an `$options` parameter to allow the theme data to be returned without theme supports.
	 *
	 * @param array $deprecated Deprecated. Not used.
	 * @param array $options {
	 *     Options arguments.
	 *
	 *     @type bool $with_supports Whether to include theme supports in the data. Default true.
	 * }
	 * @return WP_Theme_JSON Entity that holds theme data.
	 */
	public static function get_theme_data( $deprecated = array(), $options = array() ) {
		return WP_Theme_JSON_Resolver::get_theme_data( $deprecated, $options );	
	}
}
```

Reply:
The above suggested code will bypass `WP_Theme_JSON_Resolver_6_0::get_theme_data()` implementation, to call the version in Core 6.1.

Reply:
But wait a minute. This won't work if the site is running < WP 6.1 with the plugin. So the entire `WP_Theme_JSON_Resolver::get_theme_data()` code needs to be copied here it seems.